### DISCUSSION TITLE
Get length of strategy matrix (or get all matrix options)

### DISCUSSION THREAD

**Discussion by billyvg** [community/community#26930] [Actions]
<p>It would be nice to be able to get the length of a matrix list. In our specific case, we use matrix runs to parallelize a suite of tests, but it requires us to configure the total number of test runner instances we will be using. e.g.</p>
<pre><code>strategy:
  matrix:
    instance: [0, 1, 2]

...

env:
  TOTAL_TEST_GROUPS: 3
</code></pre>
<p>It would be nice to be able to either get the entire <code>instance</code> list and/or be able to get the length of it.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/billyvg">@billyvg</a> ,</p>
<p>Currently, we have no methods to calculate the number of the jobs in a matrix.</p>
<p>I checked the  <strong>matrix context</strong> , no property for the number of the jobs. I also tried some APIs, but none of them is helpful.</p>
<p>I recommend you directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by billyvg** [comment, marked as answer]
<p>So you can get the total number of jobs with <code>${{ strategy.job-total }}</code> and  the current job with <code>${{ strategy.job-index }}</code></p>
